compiler/p/codegen/OMRInstOpCodeProperties.hpp

```diff
@@ -12166,7 +12176,7 @@
    /* .name        = */ "label",
    /* .description =    "Destination of a jump", */
    /* .opcode      = */ 0x00000000,
-   /* .format      = */ UNKNOWN_FORMAT,
+   /* .format      = */ FORMAT_NONE,
    /* .minimumALS  = */ TR_Processor::TR_DefaultPPCProcessor,
    /* .properties  = */ PPCOpProp_None,
    },
```

Comment:
you seemed solely encoding-purpose focused when designating an instruction's format, even deviating from ISA's format categories. I can certainly be enlightened to understand the seemingly catch-all format of FORMAT_UNKNOWN, also TR_DefaultPPCProcessor as well.  hope this effort helps new comers understand the encoding easier ...

Reply:
For the time being, `FORMAT_UNKNOWN` is being used as a format for instructions that the new binary encoder should not attempt to encode and will always trigger an assertion failure if the new binary encoder encounters them. Currently, `FORMAT_UNKNOWN` is being used for instructions that fit one of the following criteria:

1. Are encoded using an instruction class that isn't being refactored as part of this PR and thus still use the old binary encoder, e.g. memory instructions. As part of later PRs, these will also be moved over and proper formats will be added.
2. Have not yet had unit tests written to check that they encode correctly. We have a lot of exotic and usued opcodes in this table and due to time constraints, unit tests have only been added for instructions that we use and a small subset of other instructions at this point. The idea here is to ensure that somebody doesn't unintentionally use an instruction for which the binary encoder has not been tested to work correctly. By using `FORMAT_UNKNOWN`, the new binary encoder will fail in a reliable manner rather than potentially silently encoding an instruction incorrectly. Before using such an instruction, unit tests should be added and the proper format should be filled in.
3. Are pseudoinstructions that should never be binary encoded using the normal binary encoding procedure, e.g. legacy control flow instructions like `idiv` or special opcodes like `vgdnop`.

As for `TR_DefaultPPCProcessor`, that's part of the minimum ISA level for each instruction that was added automatically via a script as part of #3930. That script used information scraped from the ISA document, so it was imperfect and used `TR_DefaultPPCProcessor` as a fallback. That's somewhat beside the point here, though, as these values are not being modified nor used as part of the binary encoding refactor.